### ISSUE TITLE
invalid error after editing dockerfile

### ISSUE THREAD

**Issue by chrisdias** [rcjsuen/dockerfile-language-server-nodejs#216]
I'm testing the Docker extension for release, with `0.0.14` of the server.

Create this dockerfile:

``` dockerfile
FROM openjdk:8-jdk-alpine
VOLUME /tmp
ARG JAVA_OPTS
ENV JAVA_OPTS=$JAVA_OPTS
ADD stickerapp.jar stickerapp.jar
EXPOSE 3000
ENTRYPOINT exec java $JAVA_OPTS -jar stickerapp.jar
# For Spring-Boot project, use the entrypoint below to reduce Tomcat startup time.
#ENTRYPOINT exec java $JAVA_OPTS -Djava.security.egd=file:/dev/./urandom -jar stickerapp.jar
```

Put the cursor at the end of the line `VOLUME /tmp` and press backspace to delete ` /tmp`
Press CMD+Z to undo your change so the line says `VOLUME /tmp` again.

Result: Red squiggles. Hover over ARG and see `Invalid error [dockerfile-utils] Unknown instruction: /tmpARG`

![image](https://user-images.githubusercontent.com/1487073/38157207-8d04d17c-3439-11e8-89e9-a146bf269172.png)

**Comment by rcjsuen** [comment]
Interesting. Thank you for the bug report, @chrisdias. I might not be able to look at it for a while pending hotel wifi...

**Comment by chrisdias** [comment]
looks like this is in 0.0.13 as well.

**Comment by rcjsuen** [comment]
I couldn't reproduce this with the [current version](https://github.com/rcjsuen/dockerfile-language-service/blob/e42bc822d782f5d8e4395147772154559277f2df/index.html) of the [online editor](https://rcjsuen.github.io/dockerfile-language-service/). I guess that makes sense. It must be something to do with VS Code instead of Monaco.

**Comment by rcjsuen** [comment]
I was able to reproduce this problem. Interestingly enough, I received two events when I initiated the undo with the second being an empty array...
```JavaScript
{
  "textDocument": {
    "uri": "file:///blah/blah/Dockerfile",
    "version": 9
  },
  "contentChanges": [
    {
      "range": {
        "start": {
          "line": 1,
          "character": 7
        },
        "end": {
          "line": 1,
          "character": 7
        }
      },
      "rangeLength": 0,
      "text": "/"
    },
    {
      "range": {
        "start": {
          "line": 1,
          "character": 8
        },
        "end": {
          "line": 1,
          "character": 8
        }
      },
      "rangeLength": 0,
      "text": "t"
    },
    {
      "range": {
        "start": {
          "line": 1,
          "character": 9
        },
        "end": {
          "line": 1,
          "character": 9
        }
      },
      "rangeLength": 0,
      "text": "m"
    },
    {
      "range": {
        "start": {
          "line": 1,
          "character": 10
        },
        "end": {
          "line": 1,
          "character": 10
        }
      },
      "rangeLength": 0,
      "text": "p"
    }
  ]
}
```
```
{
  "textDocument": {
    "uri": "file:///blah/blah/Dockerfile",
    "version": 9
  },
  "contentChanges": []
}
```

**Comment by rcjsuen** [comment]
This is caused by the fix for #58. It seems like we'll need to figure out how to merge and sort the received `contentChanges` better.

**Comment by rcjsuen** [comment]
> This is caused by the fix for #58. It seems like we'll need to figure out how to merge and sort the received contentChanges better.

Actually, I think I've just been misinterpreting the specification this whole time. I've always applied the changes in reverse order but I don't remember why. However, given that the [specification](https://github.com/Microsoft/language-server-protocol/blob/22daa79585999544a0081b3c1fa36c301e6cc8c3/specification.md) states:
```TypeScript
	/**
	 * The actual content changes. The content changes describe single state changes
	 * to the document. So if there are two content changes c1 and c2 for a document
	 * in state S10 then c1 move the document to S11 and c2 to S12.
	 */
	contentChanges: TextDocumentContentChangeEvent[];
```
It does sound like the changes should just be applied in the order it appears in the notification.

**Comment by rcjsuen** [comment]
@chrisdias Thanks again for the bug report! I've fixed the bug now and it will be included in the next release!
